pkg/controller/windowsmachine/windows/windows.go

```diff
@@ -339,41 +360,50 @@ func (vm *windows) initializeBootstrapperFiles() error {
 }
 
 // createService creates the service on the Windows VM
-func (vm *windows) createService(svc service) error {
-	out, err := vm.Run("sc.exe create "+svc.Name()+" binPath=\""+svc.BinaryPath()+" "+
-		svc.Args()+"\" start=auto", false)
+func (vm *windows) createService(svc *service) error {
+	out, err := vm.Run("sc.exe create "+svc.name+" binPath=\""+svc.binaryPath+" "+
+		svc.args+" start=auto", false)
 	if err != nil {
 		return errors.Wrapf(err, "failed to create service with output: %s", out)
 	}
 	return nil
 }
 
 // stopService stops the service that was already running
-func (vm *windows) stopService(svc service) error {
-	out, err := vm.Run("sc.exe stop "+svc.Name(), false)
+func (vm *windows) stopService(svc *service) error {
+	out, err := vm.Run("sc.exe stop "+svc.name, false)
 	if err != nil {
-		return errors.Wrapf(err, "failed to stop %s service with output: %s", svc.Name(), out)
+		return errors.Wrapf(err, "failed to stop %s service with output: %s", svc.name, out)
 	}
 	return nil
 }
 
+// serviceExists check if the given service exists on Windows VM
+func (vm *windows) serviceExists(serviceName string) bool {
+	_, err := vm.Run("sc.exe qc "+serviceName, false)
+	if err != nil {
+		return false
```

Comment:
You need to check if the error was a code 1060 and only then return false, else return error.

Reply:
I thought that was necessary because we weren't checking for service existence before querying the service for running status and that was causing the reconcile errors, hence we want to check the error code it is throwing. This is an alternative approach where I am using qc to check if service exists and query to check for running status. I can log the error here before returning false.

Reply:
its important that we make the distinction between `sc.exe qc` returning error from the service not existing, and it returning error because of various other reasons. If the service exists and we have a transient ssh error, we will think the service doesnt exist when it actually does.

Reply:
@mansikulkarni96 If you go down this route we have to be very careful about parsing the errors. Just looking for `1060` is not sufficient. Then you need to return an `error` in addition to the `bool`.